geps/gep-3792/index.md

```diff
@@ -246,16 +233,254 @@ guarantee that **all four** of the problems listed above need resolutions, and
+other. The simplest solution to this problem is to add a _trust bundle_ to the
+Gateway resource, and to create a Mesh resource which will also have a trust
+bundle:
+
+- The trust bundle in the Gateway resource will define the CA certificate(s)
+  that the OCG should trust when communicating with meshed workloads in the
+  cluster.
+
+- The trust bundle in the Mesh resource will define the CA certificate(s)
+  that the mesh should trust when communicating with the OCG.
+
+This is a straightforward way to permit each component to verify the identity
+of the other, which will provide sufficient basis for verifying identity when
+mTLS meshes are involved.
+
+- An alternative would be to define a single trust bundle, requiring the OCG
+  and the mesh to each use the same CA certificate. This adds considerable
+  operational complexity (especially in the world of enterprise PKI) without
+  any real benefit.
+
+### Solving the Protocol Problem
+
+The protocol problem is that the OCG needs a way to indicate to the mesh that
+it intends to participate in the mesh for a given connection, and the mesh
+needs to accept the OCG's participation.
+
+As a starting point for OCG/mTLS mesh interaction:
+
+- The OCG MUST use an mTLS connection to communicate with meshed workloads.
+
+- The OCG MUST use an mTLS certificate ultimately signed by a certificate in
+  the trust bundle provided to the mesh.
+
+- The mesh MUST use an mTLS certificate ultimately signed by a certificate in
+  the trust bundle provided to the OCG.
+
+- The OCG MUST send the `ocg.gateway.networking.k8s.io/v1` ALPN protocol
+  during mTLS negotiation. The mesh MUST interpret this ALPN selection as a
+  signal that the OCG intends to participate in the mesh.
+
+- The OCG MUST NOT send any additional information in the data stream before
+  client data. (This is a contrast from e.g. Linkerd's default behavior, and
+  has implications for the outbound behavior problem.)
+
+This will (obviously) require implementation work on the part of both the OCG
+implementations and the mesh implementations, so it's worth looking at some
+alternatives.
+
+- The OCG could be required to exactly mimic the ALPN/transport metadata
+  combination used by an existing mesh. However, the existing meshes don't
+  share a single common mechanism, so this would require a lot of work on the
+  OCG's part, and it wouldn't be portable between meshes.
+
+- The OCG could simply skip ALPN, and hand a "bare" mTLS connection to the
+  mesh. In general, existing meshes don't support this in the way that we
+  want: depending on the destination of the connection, they may interpret it
+  as application-level mTLS that they should treat as an opaque data stream,
+  or they may simply refuse it. This alternative, therefore, both shifts the
+  entire burden of implementation to the meshes, and probably makes it
+  impossible to correctly handle application-level mTLS.
```

Comment:
just wanted to mention that GKE Mesh is a notable exception. We use simply use bare mtls connection. And yes, we're subject to the downsides mentioned above. But the upside of interoperability between various GCP (off-GCP) data planes with GKE Mesh workloads using standard MTLS was/is deemed to be more important (i.e don't run into the protocol problem).

my intention with this comment is NOT to say custom protocol shouldn't be supported, but to point out that there are implementations that just use bare/standard mtls.